File: app/src/main/java/com/worldventures/dreamtrips/social/ui/friends/model/converter/FriendCandidateToUserConverter.java
package com.worldventures.dreamtrips.social.ui.friends.model.converter;

import com.worldventures.core.converter.Converter;
import com.worldventures.core.model.User;
import com.worldventures.dreamtrips.api.friends.model.FriendCandidate;

import io.techery.mappery.MapperyContext;

public class FriendCandidateToUserConverter implements Converter<FriendCandidate, User> {
   @Override
   public Class<FriendCandidate> sourceClass() {
      return FriendCandidate.class;
   }

   @Override
   public Class<User> targetClass() {
      return User.class;
   }

   @Override
   public User convert(MapperyContext mapperyContext, FriendCandidate apiUser) {
      User user = new User();
      user.setId(apiUser.id());
      user.setFirstName(apiUser.firstName());
      user.setLastName(apiUser.lastName());
      user.setUsername(apiUser.username());
      user.setAvatar(mapperyContext.convert(apiUser.avatar(), User.Avatar.class));
      user.setBadges(apiUser.badges());
      user.setRelationship(mapperyContext.convert(apiUser.relationship(), User.Relationship.class));
      user.setLocation(apiUser.location());
      user.setCompany(apiUser.company());
      user.setMutualFriends(mapperyContext.convert(apiUser.mutuals(), User.MutualFriends.class));
      return user;
   }
}

File: app/src/main/java/com/worldventures/dreamtrips/social/service/users/base/model/converter/FriendCandidateToUserConverter.kt
package com.worldventures.dreamtrips.social.service.users.base.model.converter

import com.worldventures.core.converter.Converter
import com.worldventures.core.model.User
import com.worldventures.dreamtrips.api.friends.model.FriendCandidate

import io.techery.mappery.MapperyContext

class FriendCandidateToUserConverter : Converter<FriendCandidate, User> {

   override fun sourceClass(): Class<FriendCandidate> = FriendCandidate::class.java

   override fun targetClass(): Class<User> = User::class.java

   override fun convert(context: MapperyContext, source: FriendCandidate) =
         User().apply {
            id = source.id()
            firstName = source.firstName()
            lastName = source.lastName()
            username = source.username()
            location = source.location()
            company = source.company()
            badges = source.badges()
            avatar = context.convert(source.avatar(), User.Avatar::class.java)
            mutualFriends = source.mutuals()?.let { context.convert(it, User.MutualFriends::class.java) }
            relationship = source.relationship()?.let { context.convert(it, User.Relationship::class.java) }
         }

}
